pyccel/parser/base.py

```diff
@@ -127,6 +127,7 @@ def __init__(self, *, headers=None, decorators=None):
         self._classes   = OrderedDict()
         self._functions = OrderedDict()
         self._macros    = OrderedDict()
+        self._templates = {}
```

Comment:
What is the reason for now using an ordered dictionary like for every other attributes?

Reply:
yes this should be an ordered dictionary as well